### PR TITLE
Add HKBU docs

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by Ruttie2006 — APPROVED** [review]

**Review by TheMulhima — APPROVED** [review]

File: IdeExtensions/BuildTools/HKBU/modpack.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+Available properties:
+- `$(OutputZipSHA256)`: SHA256 of the output publish zip
+
+## Configuration
+
+- `$(ExportDir)`: Specifies the directory for the packaged output. Defaults to "$(ProjectDir)\bin\Publish\"
+- `$(PublishZipPath)`: Specify the save path of the ZIP output after packaging. Defaults to "$(ExportDir)\Publish.zip"
+- `$(ReleaseInfoPath)`: Specifies the output location of the ReleaseInfo information constructed after packaging. Currently, only SHA256 is included. Default is "\$([System.IO.Path]::GetDirectoryName('$(PublishZipPath)'))\ReleaseInfo.txt"
```
Comment (Ruttie2006) [inline IdeExtensions/BuildTools/HKBU/modpack.md:44]: Maybe change it to:
`$(ReleaseInfoPath)`: Specifies the output location of the ReleaseInfo information constructed after packaging. Currently, only SHA256 is included. Default is "$(PublishZipPath)\ReleaseInfo.txt"

This way it's less confusing.

File: IdeExtensions/BuildTools/HKBU/monomod-hooks-helper.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
+title: Mono Mod Hooks Helper
+nav_order: 2
+parent: Hollow Knight Build Utils
+---
+
+Allow developers to use MonoModHooks outside of `MMHOOK_Assembly-CSharp` and `MMHOOK_PlayMaker.dll`
```
Comment (Ruttie2006) [inline IdeExtensions/BuildTools/HKBU/monomod-hooks-helper.md:7]: MonoMod hooks*

File: IdeExtensions/BuildTools/HKBU/hkbu.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+```
+or
+```
+<PackageReference Include="HKBuildUtils" Version="*">
+  <PrivateAssets>all</PrivateAssets>
+  <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+</PackageReference>
+```
+
+## Migrated from mod template made by BadMagic
```
Comment (Ruttie2006) [inline IdeExtensions/BuildTools/HKBU/hkbu.md:42]: Migrating from the mod template made by BadMagic*

**Review by Ruttie2006 — COMMENTED** [review]
Some small things.
This is mainly for future reference, I'll fix this myself.

**Pull request merged**